### PR TITLE
[bugfix] MaterializedPostgreSQL Cannot  attach table  when pg dbname  contains "-", need doubleQuoting

### LINKED ISSUE #62691: ATTACH TABLE not working for MaterializedPostgreSQL where postgres table contains a "-"
**Describe what's wrong**

Reproducable steps:
1. Create Postgres db with a "-", for instance: "example-db"
2. Create `MaterializedPostgreSQL` according to https://clickhouse.com/docs/en/engines/database-engines/materialized-postgresql#settings
3. Attach new table like: `ATTACH TABLE clickhouse_example_db.new_table`;
4. Error: 
```
SQL Error [619] [07000]: Code: 619. DB::Exception: Failed to add table `new_table` to replication. Info: std::exception. Code: 1001, type: pqxx::syntax_error, e.what() = ERROR:  syntax error at or near "-"
LINE 1: ALTER PUBLICATION example-db_ch_publication ADD TA...
                                    ^
 (version 24.4.1.1097 (official build)). (POSTGRESQL_REPLICATION_INTERNAL_ERROR) (version 24.4.1.1097 (official build))
, server ClickHouseNode [uri=http://localhost:8123/default, options={use_server_time_zone=false,use_time_zone=false}]@1017777186
````

**Does it reproduce on the most recent release?**

Yes, on latest version 24.4.1.1097

**Expected behavior**

Expected the table to be added to the MaterializedPostgreSQL database. This works when the source db does not have a "-"

### LINKED ISSUE #62827: MaterializedPostgreSQL: failed ATTACH TABLE (publication name is not escaped)
Following up from closed #55251 

**Describe the unexpected behaviour**

Database name with a hyphen is not escaped, during the `attach` command.
The original `create database` is executed without issues with data is replicated, but once you try to add more tables, you are hit with an error due to the `-` in the db name.

**How to reproduce**
* Which ClickHouse server version to use
24.3.2.23 
* Which interface to use, if matters
cli
* Queries to run that lead to unexpected result
```
CREATE DATABASE test_sync_materialized_db_engine
ENGINE = MaterializedPostgreSQL('db-postgresql:21234', 'test-sync', 'admin', 'xxxxxx')
SETTINGS materialized_postgresql_tables_list = 'test_Table1,Table2';
```

```
ATTACH TABLE test_sync_materialized_db_engine.Table3

Query id: 1e169c7d-f92c-4d43-9fe2-5384efa9faf3


Elapsed: 0.177 sec. 

Received exception from server (version 24.3.2):
Code: 619. DB::Exception: Received from localhost:9000. DB::Exception: Failed to add table `Table3` to replication. Info: std::exception. Code: 1001, type: pqxx::syntax_error, e.what() = ERROR:  syntax error at or near "-"
LINE 1: ALTER PUBLICATION test-sync_ch_publication ADD TABLE ONLY...
                                 ^
 (version 24.3.2.23 (official build)). (POSTGRESQL_REPLICATION_INTERNAL_ERROR)

```

**Expected behavior**
`ATTACH TABLE` should not fail and it should create a new publication.

### PR DESCRIPTION
**Pull request by takakawa**
Hi, I found some code missing doubleQuoting publication_name, thus causing a very strange behavior.
One Can successufly create publication_name because is is doubleQuoted already.
But One Can not alter publication_name because it is not doubleQuoted.

I tested the code in our prod env , it works  as desired. I can attach new table now.

Closes https://github.com/ClickHouse/ClickHouse/issues/62827. Closes https://github.com/ClickHouse/ClickHouse/issues/62691.

### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix attaching table when pg dbname contains "-" in MaterializedPostgreSQL.

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

---
### Modify your CI run:
**NOTE:** If your merge the PR with modified CI you **MUST KNOW** what you are doing
**NOTE:** Checked options will be applied if set before CI RunConfig/PrepareRunConfig step

#### Include tests (required builds will be added automatically):
- [ ]  Fast test
- [x]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Unit tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with Analyzer
- [ ]  Add your option here
- [x]  Woolen Wolfdog

#### Exclude tests:
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Stress tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with MSAN
- [ ]  All with UBSAN
- [ ]  All with Coverage
- [ ]  All with Aarch64
- [ ]  Add your option here

### CODE REVIEW

**Review by nikitamikhaylov — APPROVED** [review]

**Review by nikitamikhaylov — CHANGES_REQUESTED** [review]
Approving was a misclick.

File: src/Storages/PostgreSQL/PostgreSQLReplicationHandler.cpp [hunk 1]
```diff
@@ -157,7 +157,7 @@ PostgreSQLReplicationHandler::PostgreSQLReplicationHandler(
 
     checkReplicationSlot(replication_slot);
 
-    LOG_INFO(log, "Using replication slot {} and publication {}", replication_slot, publication_name);
+    LOG_INFO(log, "Using replication slot {} and publication {}", replication_slot, doubleQuoteString(publication_name));
```
Comment (nikitamikhaylov) [inline src/Storages/PostgreSQL/PostgreSQLReplicationHandler.cpp:160]: Why do we need double quotes? Why not to use single quotes or backticks?

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit 5a6fe87b7c8902bcef6be355f157bc07a1fa0182 with description of existing statuses. It's updated for the latest CI running*

[❌ Click here](https://s3.amazonaws.com/clickhouse-test-reports/62730/5a6fe87b7c8902bcef6be355f157bc07a1fa0182/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td><a href="https://github.com/ClickHouse/ClickHouse/blob/master/.github/PULL_REQUEST_TEMPLATE.md?plain=1">❌ failure</a></td></tr>
<tbody>
</table>

**Review by kssenii — COMMENTED** [review]
Could you please try to add a test for this case in https://github.com/ClickHouse/ClickHouse/blob/master/tests/integration/test_postgresql_replica_database_engine_2/test.py ? Running a test can be done via `./runner  --   test_postgresql_replica_database_engine_2/test.py::test_name`. A test with 'ATTACH TABLE' to MaterializedPostgreSQL database should be enough to test the bug.

File: src/Storages/PostgreSQL/PostgreSQLReplicationHandler.cpp (hunk 1 above)
Comment (kssenii) [reply src/Storages/PostgreSQL/PostgreSQLReplicationHandler.cpp:160]: For logging it does not matter what we use, but for query to postgres we should use double quotes.

Comment (takakawa) [reply src/Storages/PostgreSQL/PostgreSQLReplicationHandler.cpp:160]: Yes, It does not matter when logging.  Using single quotes or backticks may be the same.
I saw other logs using doubleQuoteString so I used it.    :-)

**Comment by takakawa** [comment]
Hadn't figured out what happened.I ran an existing test and  it failed. 

`$ sudo ./runner  --   test_postgresql_replica_database_engine_2/test.py::test_add_new_table_to_replication
`

`ERROR test_postgresql_replica_database_engine_2/test.py::test_add_new_table_to_replication - Exception: Timed out while waiting for instance `instance' with ip address 172.16.1.4 to start. Container status: running, logs:
======================================================================= 1 error in 313.35s (0:05:13) =======================================================================
Traceback (most recent call last):
  File "./runner", line 437, in <module>
    subprocess.check_call(cmd, shell=True)
  File "/usr/lib64/python3.6/subprocess.py", line 311, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command 'docker run --net=host -it --rm --name clickhouse_integration_tests_a8j1bw --privileged --dns-search='.' --volume=/usr/bin/clickhouse-odbc-bridge:/clickhouse-odbc-bridge --volume=/usr/bin/clickhouse:/clickhouse --volume=/usr/bin/clickhouse-library-bridge:/clickhouse-library-bridge --volume=/home/gaochuan/ClickHouse2/programs/server:/clickhouse-config --volume=/home/gaochuan/ClickHouse2/tests/integration:/ClickHouse/tests/integration --volume=/home/gaochuan/ClickHouse2/utils/backupview:/ClickHouse/utils/backupview --volume=/home/gaochuan/ClickHouse2/utils/grpc-client/pb2:/ClickHouse/utils/grpc-client/pb2 --volume=/run:/run/host:ro --volume=clickhouse_integration_tests_volume:/var/lib/docker   -e DOCKER_CLIENT_TIMEOUT=300 -e COMPOSE_HTTP_TIMEOUT=600  -e PYTHONUNBUFFERED=1 -e PYTEST_ADDOPTS=" test_postgresql_replica_database_engine_2/test.py::test_add_new_table_to_replication   -vvv" clickhouse/integration-tests-runner:latest ' returned non-zero exit status 1.`

**Review by kssenii — APPROVED** [review]

**Review by nikitamikhaylov — APPROVED** [review]

**Comment by flemzord** [comment]
Any news on this PR? We have the same problem on our side.

**Pull request merged**